### PR TITLE
Add LoanTime class

### PR DESCRIPTION
**Pull request by prokarius**
Add LoanTime class

### CODE REVIEW

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    /**
+     * Constructs a {@code LoanTime} that is now.
+     */
+    public LoanTime() {
+        value = Instant.now();
+    }
+
+    /**
+     * Constructs an {@code LoanTime}.
+     * Note this is a private constructor
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: Right, my mistake.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    /**
+     * Returns the difference in time given one other LoanTime object.
+     * This will be returned as a long of number of second.
+     * Function returns 0 if the specified other LoanTime is before the current LoanTime.
+     *
+     * @param otherTime LoanTime object to be compared to.
+     */
+    public long loanTimeDifferenceMinutes(LoanTime otherTime) {
+        long timeDifference = (otherTime.value.toEpochMilli() - this.value.toEpochMilli());
+        return (timeDifference > 0) ? timeDifference / 60000 : 0;
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: If the time difference == 0, the returned value is going to be zero also.
I'll change the comparator to be consistent with the description.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    /**
+     * Returns the DDMMYYYY representation of the stored LoanTime.
+     * The format returned is "YYYY-MM-DD, HH:MM".
+     */
+    @Override
+    public String toString() {
+        // We create a new date object and return the DDMMYYYY representation of it
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: This comment refers to how we will be generate the string representation.

I will capitalize "Date" to reflect the new object being created.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+        // Check and append the time
+        checkArgument(isValidTime(loanTime), MESSAGE_TIME_CONSTRAINTS);
+        stringBuilder.append(loanTime);
+
+        // Append the "Z" and other padding.
+        stringBuilder.append(":00.00Z");
+
+        // Find a way to parse the input date correctly...
+        // I hardcoded in the -8 Hours GMT into here.
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: OK. Edited to reflect the change.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+    /*
+     * 2 versions of the regex:
+     * A long version where a date is specified
+     * Format: YYYY-MM-DD HH:mm
+     *
+     * A short version where only the time is specified
+     * Format HH:MM
+     */
+    public static final String LONG_LOANTIME_VALIDATION_REGEX = "\\d{4}-\\d{2}-\\d{2} +\\d{2}:\\d{2}";
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: These will all be done within the validation tests. I feel that the regex is there to filter out unparseable inputs, and the validity checks can be done within the respective `isValid` methods.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+        // Check and append the time
+        checkArgument(isValidTime(loanTime), MESSAGE_TIME_CONSTRAINTS);
+        stringBuilder.append(loanTime);
+
+        // Append the "Z" and other padding.
+        stringBuilder.append(":00.00Z");
+
+        // Find a way to parse the input date correctly...
+        // I hardcoded in the -8 Hours GMT into here.
```
Comment (wn96) [inline src/main/java/seedu/address/model/loan/LoanTime.java]: Please edit the comment.

File: src/main/java/seedu/address/model/loan/LoanTime.java [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        return true;
+    }
+
+    // TODO
+    /**
+     * Returns if a given string is a valid Time.
+     */
+    public static boolean isValidTime(String test) {
+        String[] timeData = test.split(":");
+        return (Integer.parseInt(timeData[0]) < 24) && (Integer.parseInt(timeData[1]) < 60);
```
[the remark this answers is not in the document]
Comment (prokarius) [reply src/main/java/seedu/address/model/loan/LoanTime.java]: I have added a layer of input validation.

**Pull request merged**